src/main/java/net/neoforged/neoforge/client/model/generators/template/RotationBuilder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.model.generators.template;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.client.renderer.block.model.BlockElementRotation;
+import net.minecraft.core.Direction;
+import net.minecraft.util.Mth;
+import org.jetbrains.annotations.Nullable;
+import org.joml.Vector3f;
+
+public final class RotationBuilder {
+    @Nullable
+    private Vector3f origin;
+    @Nullable
+    private Direction.Axis axis;
+    private float angle;
+    private boolean rescale;
+
+    public RotationBuilder origin(float x, float y, float z) {
```

Comment:
This method should have a doc comment explaining that this defines the point around which the element will be rotated.